src/libaktualizr/package_manager/packagemanagerinterface.h

```diff
@@ -35,47 +35,24 @@ enum class TargetStatus {
 
 class PackageManagerInterface {
  public:
-  PackageManagerInterface(PackageConfig pconfig, BootloaderConfig bconfig, std::shared_ptr<INvStorage> storage,
```

Comment:
As you mentioned in your comment, this would break #1498. Do you then have an alternative design to propose? Also, the exception in this PR is the Debian package manager which we don't really care about.

Maybe an intermediate solution would just be to pass a const reference to the full config here, so that implementations can pick what's needed.

In any case, I would prefer if we kept this change in another PR since it's independent from the other changes.

Reply:
It's not independent, the interface cleanup depends on the fix implementation.

I thought that this is what Eugene intended eventually to do with moving the bootloader instance under the ostree package manager.

I will remove the pack man interface cleanup from this PR and create another PR, although the later will depend on this PR.